docs/networking.md

```diff
@@ -0,0 +1,262 @@
+    - App Subnet: /19
+    - Data Subnet: /20
+
+  - **Shared Subnets Between Workloads:**
+    - Subnets are shared among all workloads within the same VPC.
+    - All workloads in the same VPC/subnet share the same IP address pool.
+      
+  - **Security Measures:**
+    - Shared subnets utilize the same IP pool to prevent workloads from communicating with each other across accounts.
+    - This design enhances security by isolating workloads within the same VPC, minimizing cross-workload communication.
+
+
+
+<!-- - Differentiation between public and private subnets, all subnets in a workload VPC are private
+- Subnets created in each workload VPC are Web, App, and Data
+- Use cases for each type of subnet
+- Configuration of subnets in each workload VPC
+  - CIDR blocks
+    - Web: /20
+    - App: /19
+    - Data: /20
+  - These subnets are shared between all workloads in a VPC
+    - Same IP pool for all workloads in the same VPC/subnet
+    - This are security measures to prevent workloads from communicating with each other across accounts -->
+
+### Security groups and NACLs
+Security Groups and Network Access Control Lists (NACLs) play distinct roles in ensuring the security of Workload VPCs, with Security Groups acting as instance-level firewalls and NACLs providing an additional layer of defense at the subnet level
+
+- **Difference Between Security Groups and NACLs:**
+  - **Security Groups:**
+    - Instance-level stateful firewalls controlling inbound and outbound traffic.
+    - Rule-based and allow-list focused, supporting ingress/egress based on protocols and sources/destinations.
+
+  - **NACLs:**
+    - Stateless constructs operating at the subnet level.
+    - Provide an additional layer of security, defining rules for inbound and outbound traffic at the subnet level.
+
+- **Role of NACLs in Subnet Security:**
+  - NACLs in ASEA act as a defense-in-depth measure at the subnet level.
+  - They are used sparingly, particularly in Data subnets, to enforce segmentation, allowing only necessary inbound traffic from App subnets within the same VPC.
+
+- **ASEA Created and Managed Security Groups and NACLs in Each Workload VPC:**
+  <!-- Refine this section -->
+  - **Security Groups:**
+    - Configured as instance-level firewalls, recommended as the primary data-plane isolation mechanism.
+    - Emphasize stateful rules, focusing on allow-listing required ingress traffic.
+
+  - **NACLs:**
+    - Used sparingly as a defense-in-depth measure, particularly in Data subnets, to enforce segmentation.
+    - ASEA encourages review and tailoring based on specific security requirements.
+
+- **Best Practices for Configuring Security Groups in Workload VPCs:**
+  <!-- Refine this section -->
+  - Security Groups are recommended as the primary data-plane isolation mechanism.
+  - Ingress rules often set to 'allow all' mode (0.0.0.0/0) for ease of operations, with emphasis on consistently allow-listing required ingress traffic.
```

Comment:
Only open the necessary ports, IP ranges/addresses and grant the minimum required permissions to each security group to minimize exposure to vulnerabilities.
```suggestion
  - Only open the necessary ports, IP ranges/addresses and grant the minimum required permissions to each security group to minimize exposure to vulnerabilities.
```